pkg/controller/autoscaling/elasticsearch/autoscaler/offline.go

```diff
@@ -39,12 +38,8 @@ func GetOfflineNodeSetsResources(
 		}
 	}
 
-	// Ensure that the min. number of nodes is in the allowed range.
-	if expectedNodeCount < autoscalingSpec.NodeCount.Min {
-		expectedNodeCount = autoscalingSpec.NodeCount.Min
-	} else if expectedNodeCount > autoscalingSpec.NodeCount.Max {
-		expectedNodeCount = autoscalingSpec.NodeCount.Max
-	}
+	// Ensure that the number of nodes is in the allowed range.
+	expectedNodeCount = autoscalingSpec.NodeCount.Adjust(expectedNodeCount)
```

Comment:
> Should this be called NodeCountRange or something like that to make clear that this is not a single integer value but a range (also in other places)

This is to be consistent with the spec, where `nodeCount`, `cpu`, `memory` or `storage` actually describe ranges:

```json
"resources": {
   "nodeCount": {
   	"min": 3, "max": 8
   },
   "cpu": {
   	"min": 2, "max": 6
   },
   "memory": {
   	"min": "6Gi", "max": "10Gi"
   },
   "storage": {
   	"min": "1Gi", "max": "4Gi"
   }
  }
```

Also if we want to be consistent we should rename the other fields:

```golang
type AutoscalingResources struct {
	CPU     *QuantityRange `json:"cpu,omitempty"`
	Memory  *QuantityRange `json:"memory,omitempty"`
	Storage *QuantityRange `json:"storage,omitempty"`

	// NodeCount is used to model the minimum and the maximum number of nodes over all the NodeSets managed by a same autoscaling policy.
	NodeCount CountRange `json:"nodeCount"`
}
```

⬇️ 

```golang
type AutoscalingResources struct {
	CPURange     *QuantityRange `json:"cpu,omitempty"`
	MemoryRange  *QuantityRange `json:"memory,omitempty"`
	StorageRange *QuantityRange `json:"storage,omitempty"`

	// NodeCountRange is used to model the minimum and the maximum number of nodes over all the NodeSets managed by a same autoscaling policy.
	NodeCountRange CountRange `json:"nodeCount"`
}
```

I don't have a strong opinion, and I'm fine to rename the fields in the code as in the snippet above ⬆️ , but I think I prefer to keep a short name in serialized version.

Reply:
Agreed that the current names work well in JSON, because it is immediately clear that we are talking about ranges from the nested `min` and `max` attributes. 

I thought it would be marginally better to have the range reflected in the name just because the range property is otherwise not always clear when reading. I also don't feel super strong about it so I would leave it up to you.